api/ingress/v1alpha1/domain_types.go

```diff
@@ -45,6 +52,11 @@ type DomainSpec struct {
 	// Region is the region in which to reserve the domain
 	// +kubebuilder:validation:Required
 	Region string `json:"region,omitempty"`
+
+	// DomainReclaimPolicy is the policy to use when the domain is deleted
+	// +kubebuilder:validation:Enum=Delete;Retain
+	// +kubebuilder:default=Delete
```

Comment:
should this default to Retain to be passive?

Reply:
The current behavior is to delete the domain, so this is passive. We just don't issue deletes on the domain CRs from the other controllers for this reason. 

Eventually, we can have a `defaultDomainReclaimPolicy` that is set on the controller when it creates the domains so that we can delete the domain CRs when reconciling ingress/gw/lb services if the reclaim policy is retain.